lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -955,138 +986,166 @@ def determine_agent_level_common_labels
       labels
     end
 
-    # Determine the group level monitored resource and common labels shared by a
-    # collection of entries.
-    def determine_group_level_monitored_resource_and_labels(tag)
-      # Determine group level monitored resource type. For certain types,
-      # extract useful info from the tag and store those in
-      # matched_regex_group.
-      group_resource_type, matched_regex_group =
-        determine_group_level_monitored_resource_type(tag)
-
-      # Determine group level monitored resource labels and common labels.
-      group_resource_labels, group_common_labels =
-        determine_group_level_labels(group_resource_type, matched_regex_group)
-
-      group_resource = Google::Apis::LoggingV2beta1::MonitoredResource.new(
-        type: group_resource_type,
-        labels: group_resource_labels.to_h
-      )
-
-      # Freeze the per-request state. Any further changes must be made on a
-      # per-entry basis.
-      group_resource.freeze
-      group_resource.labels.freeze
-      group_common_labels.freeze
-
-      [group_resource, group_common_labels]
+    # Group the log entries by tag and locally_unique_id pairs.
+    def group_log_entries_by_tag_and_locally_unique_id(chunk)
+      groups = {}
+      chunk.msgpack_each do |tag, time, record|
+        sanitized_tag = sanitize_tag(tag)
+        if sanitized_tag.nil?
+          @log.warn "Dropping log entries with invalid tag: '#{tag}'. " \
+                    'A tag should be a string with utf8 characters.'
+          next
+        end
+        locally_unique_id = record.delete(DEFAULT_UNIQUE_ID_KEY) if
+          record.is_a?(Hash)
+        hash_key = [sanitized_tag, locally_unique_id].freeze
```

Comment:
Ah, I removed it in order to leave it as `nil` so that later on we can evaluate `if local_resource_id` as false. If I convert it to an empty string, it becomes confusing as we also have the implicit local resource id as an empty string.